### PR TITLE
Add initial CNI prow jobs

### PR DESCRIPTION
**Pull request by howardjohn**
There are some other tests as well but they are really complicated, I want to get these in then can wrap my head around what they are doing.

I had to make a couple changes to the config generation code since it was dependent on istio/istio.

### CODE REVIEW

**Comment by howardjohn** [comment]
I will turn on tide once I get all migrate over and make sure tests pass ok

**Review by geeknoid — APPROVED** [review]

**Comment by howardjohn** [comment]
/lgtm

Owners file was updated.. should be approved

**Comment by howardjohn** [comment]
/assign @utka 
Can you approve this? Me and Martin should be owners now but looks like it doesn't take effect

**Comment by cjwagner** [comment]
> Owners file was updated.. should be approved

> Can you approve this? Me and Martin should be owners now but looks like it doesn't take effect

You are not in the necessary OWNERS file: https://github.com/istio/test-infra/blob/master/OWNERS
See the approval comment for an explanation and links to the least privileged (deepest directory level) OWNERS files that still required for approval: https://github.com/istio/test-infra/pull/1575#issuecomment-517788189

**Comment by howardjohn** [comment]
Ah my bad, I was thinking it needed owners for https://github.com/istio/test-infra/blob/master/prow/config/OWNERS not the CNI test folder.

**Comment by cjwagner** [comment]
I see you were added as approvers of `prow/config/`, but this file also modifies a file under `prow/cluster/jobs/`. I recommend adding a `prow/cluster/jobs/OWNERS` file and more OWNERS files at deeper directories to delegate job config management to job/repo owners.

**Review by fejta — APPROVED** [review]
/hold

File: prow/cluster/jobs/istio/cni/istio.cni.master.yaml
Comment (geeknoid) [inline prow/cluster/jobs/istio/cni/istio.cni.master.yaml:9]: Do we need this to be constrained to the master branch, or should this job config apply to all branches?

Comment (howardjohn) [reply prow/cluster/jobs/istio/cni/istio.cni.master.yaml:9]: we could make it all branches, the problem is eventually we may add some incompatible changes and need to fork them off. Given cni tests don't have as much churn as Istio/Istio ones, doing all branches is probably pretty safe though

**Review by fejta — APPROVED** [review]
/hold

**Comment by howardjohn** [comment]
/hold cancel

I'll make it for other branchs in a follow up, need to make it required and add the other tests still anyways, just want to get this started

**Pull request merged**
